src/ovp_pipeline/live_concept_agent.py

```
@@ -0,0 +1,459 @@
+    first balanced ``{...}`` block.  Raises :class:`ValueError`
+    when no JSON object can be found at all.
+
+    Missing keys default to empty strings — partial output is
+    valid (an LLM that updates only the synthesis section without
+    touching tensions is a fine outcome).
+    """
+    if not response_text:
+        raise ValueError("empty response from agent")
+    match = re.search(r"\{.*\}", response_text, re.DOTALL)
+    if not match:
+        raise ValueError("no JSON object found in agent response")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)